util/src/main/java/io/kubernetes/client/util/ModelMapper.java

```diff
@@ -368,43 +378,63 @@ private static void initModelMap() throws IOException {
   }
 
   static class BiDirectionalMap<K, V> {
-    private Map<K, V> kvMap = new HashMap<>();
-    private Map<V, K> vkMap = new HashMap<>();
-    private Striped<ReadWriteLock> lock =
-        Striped.readWriteLock(Runtime.getRuntime().availableProcessors() * 4);
+
+    private Map<K, V> kvMap = new ConcurrentHashMap<>();
+
+    private Map<V, K> vkMap = new ConcurrentHashMap<>();
```

Comment:
What is the locking in here? We used Striped so that there would be multiple locks to reduce contention, I'd rather not go back to a single lock.

Reply:
`ConcurrentHashMap` has striped locks internally. I didn't see any tests (I know they are hard to write) or comments about the implementation, so I'm not sure what the race condition is that we are trying to avoid. Can you elaborate, and maybe we can improve it?

Reply:
ok, I'm happy if it has striped locks

There's no race condition to avoid, simply that the bi-directional map is used to store API -> type mappings that are used in a number of different places, and we want to reduce contention on that lookup if there are multiple threads all performing lookups at the same time.